### DISCUSSION TITLE
renovate auto-merge

### DISCUSSION THREAD

**Discussion by dirien** [SchwarzIT/go-template#50] [Ideas]
Hi everyone,

i wanted to talk about the current renovate settings in `go-template`

      "matchUpdateTypes": ["major", "minor", "patch", "pin", "digest"],
      "automerge": true

I would suggest to remove `major` from the automerge? What are your thoughts?

Source:
https://docs.renovatebot.com/configuration-options/#matchupdatetypes

**Comment by brumhard** [comment]
Hi there,

thanks for taking so much care about our renovate settings 👍🏻 
We discussed this as well and came to the conclusion that it's ok to also update major versions as long as the CI tests go through successfully.
Of course you could also argue that we're maybe to optimistic with our test quality 😄 
What do you think?

**Reply by dirien** [reply]
Fine, with this. 

As long all the maintainer know, that a broken main can't be handled by the "Verursacher" principle and someone has to take care to fix. 

Otherwhise we have a broken main branch which could stop others from work. Nobody wants to fork a broken branch or merge into a broken one.

**Reply by brumhard** [reply]
Does the automerge actually overwrite the PR checks like linting and testing and just merges instantly? If so we should probably disable automerge overall.

**Reply by dirien** [reply]
You need to set `automergeType` to branch. Currently its on `pr`


> If you prefer that Renovate more silently automerge without Pull Requests at all, you can configure "automergeType": "branch". In this case Renovate will:
> 
> Create the branch, wait for test results
> Rebase it any time it gets out of date with the base branch
> Automerge the branch commit if it's: (a) up-to-date with the base branch, and (b) passing all tests
> As a backup, raise a PR only if either: (a) tests fail, or (b) tests remain pending for too long (default: 24 hours)

https://docs.renovatebot.com/configuration-options/#automergetype

**Reply by brumhard** [reply]
But I think PR should be the right one in this case:
> Automerging defaults to using Pull Requests (automergeType="pr"). In that case Renovate first creates a branch and associated Pull Request, and then automerges the PR on a subsequent run once it detects the PR's status checks are "green".

In that way all the tests that are triggered on PRs should automatically run and if one fails it won't automerge if I understand correctly. And like this the main branch should always be in a valid state, shouldn't it?

**Reply by dirien** [reply]
Yes, reading the chapter again: `pr` is right! `branch`is for more silent merging, without a PR.
